### ISSUE TITLE
FinishBenchmark() called twice in bleed

### ISSUE THREAD

**Issue by anvilvapre** [OpenRA/OpenRA#18753]
bleed, up to commit 5e42c03afce14580399ac25612106c8cee38f50d

Game calls FinishBenchmark() twice at game end - during benchmark.

OpenRA.Game/World.cs
614:                    Game.FinishBenchmark();

OpenRA.Game/Game.cs
170:                    OrderManager.World.GameOver += FinishBenchmark;

Causing benchmark column names and records to be written twice to the same log.

**Comment by anvilvapre** [comment]
Only the first occurrence - in World.cs - causes benchmark samples to be written. Causing the results to be written twice. Once for the world Type Regular and the second time for the world type ShellMap.

When commenting out Game.FinishBenchmark(); in OpenRA.Game/World.cs at 614 no benchmark results are written at all. The GameOver order event in Game.cs is never fired (when running a replay map).

Possible solution:
a. Only call Game.FinishBenchmark() in World.cs when world type is not shell map.
b. Remove Game.cs line 170, as currently it never is called.

Why fix?
- Tools like Excel and pandas break on reading the CVS because some row again contains the header column string values causing the type of the columns to be interpreted as string.

**Comment by Mailaender** [comment]
Somehow related: https://github.com/OpenRA/OpenRA/issues/18725

**Comment by teinarss** [comment]
Yeah i have know about this problem, but haven't had any incitement for fixing it, mainly because i'm the only one that have been using this feature and locally I have just applied a quick fix when doing benchmarks.

**Comment by anvilvapre** [comment]
`player.World.EndGame();` is never called in mp games.
wouldn't know how to fix it, so i will leave it to you/others.
